engine/flink/components/kafka/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/FlinkKafkaComponentProvider.scala

```diff
@@ -64,29 +65,29 @@ class FlinkKafkaComponentProvider extends ComponentProvider {
 
   override def isAutoLoaded: Boolean = false
 
-  private def dependenciesWithDisabledNamespacingIfApplicable(
+  private def modelConfigWithDisabledNamespacingIfApplicable(
       config: Config,
-      dependencies: ProcessObjectDependencies
-  ): ProcessObjectDependencies = {
+      modelConfig: ModelConfig
+  ): ModelConfig = {
     val disableNamespacePath = "disableNamespace"
     if (config.hasPath(disableNamespacePath) && config.getBoolean(disableNamespacePath)) {
-      dependencies.copy(namingStrategy = NamingStrategy.Disabled)
+      modelConfig.copy(namingStrategy = NamingStrategy.Disabled)
     } else {
-      dependencies
+      modelConfig
     }
   }
 
 }
 
-//FIXME: Kafka components should not depend directly on ProcessObjectDependencies, only on
+//FIXME: Kafka components should not depend directly on ModelConfig, only on
 //appropriate config, this class is temporary solution, where we pass modified dependencies
 private[transformer] object TemporaryKafkaConfigMapping {
 
-  def prepareDependencies(config: Config, dependencies: ProcessObjectDependencies): ProcessObjectDependencies = {
+  def prepareModelConfig(config: Config, modelConfig: ModelConfig): ModelConfig = {
     val kafkaConfig = config.getConfig("config")
     val kafkaConfigMergedWithGlobalConfig =
-      dependencies.config.withValue(KafkaConfig.DefaultGlobalKafkaConfigPath, fromAnyRef(kafkaConfig.root()))
-    ProcessObjectDependencies.withConfig(kafkaConfigMergedWithGlobalConfig)
+      modelConfig.underlyingConfig.withValue(KafkaConfig.DefaultGlobalKafkaConfigPath, fromAnyRef(kafkaConfig.root()))
```

Comment:
Maybe it would be nice to add some util method in ModelConfig companion object? It would replace those 2 lines - by creating new ModelConfig based on old ModelConfig and some underlying config transformation Config => Config.